pandas/core/indexes/multi.py

```diff
@@ -1707,7 +1707,7 @@ def remove_unused_levels(self):
                 # ... and reassigned value -1:
                 code_mapping[uniques] = np.arange(len(uniques)) - has_na
 
-                level_codes = code_mapping[level_codes]
+                level_codes = code_mapping[level_codes].astype(int)
```

Comment:
why would u need to do this?

Reply:
This is necessary given the update to `coerce_indexer_dtype` (it breaks some tests of this method). I wasn't sure about this change, I mentioned it in the initial report. The issue could be handled more locally in the `MultiIndex._set_codes`. But it seems to me you don't really want to pass `floats` to `categorical` indexers (which is happening here). Looking at it now, we should probably do `code_mapping[uniques] = np.arange(len(uniques), dtype=int)` above.

Reply:
I don't want to interject a random astype in a particular method; this copies, and it is very fragile

Reply:
I've replace this with `code_mapping[uniques] = np.arange(len(uniques), dtype=int) - has_na` which I should have done initially. This seems to natural to me as ensure `code_mapping` is of the correct type to begin with.